### ISSUE TITLE
How to update from 5.0.6 to 5.1.0

### ISSUE THREAD

**Issue by FlavioSuarez** [abydahana/aksara#468]
Hi,

There are instructions on how to update from 5.0.6 to 5.1.0?
I ran composer update and it just updated some dependencies...
I deleted the composer.lock file and got the same results...
I tried with composer upgrade and same results...
Trying it on local environment...

Thanks for your help!

**Comment by abydahana** [comment]
Hello,
A major updates can only be done manually. You can override the following file into your old Aksara version:
```
/aksara/*
/public/assets/*
/public/index.php
/composer.json
```
Run `composer update` after overriding these files.

See detailed information about affected file changes: https://github.com/abydahana/aksara/compare/5.0.6...5.1.0

Make sure to backup your current project.

**Comment by FlavioSuarez** [comment]
OK, just to be sure about the right way to update.
Thanks!

**Comment by FlavioSuarez** [comment]
It almost worked!
I just had to change the config.php file, 
- From define('DB_PORT', '3306');
- To define('DB_PORT', 3306);

**Comment by abydahana** [comment]
> It almost worked! I just had to change, in the config.php file, the PORT setting from string to int
> 
> * From define('DB_PORT', '3306');
> * To define('DB_PORT', 3306);

Does the string port didn't working anymore?

**Comment by FlavioSuarez** [comment]
Copying the 5.1.0 files over the 5.0.6 install, nope... 

I get the following error:

CodeIgniter\Database\Exceptions\DatabaseException
Unable to connect to the database.
Main connection [MySQLi]: mysqli::real_connect(): Argument #5 ($port) must be of type ?int, string given 

But maybe on a fressh install this don't happens...

**Comment by FlavioSuarez** [comment]
Additionally, I see that now the default environment is "development"... Where I can set it to Production?

**Comment by abydahana** [comment]
Okay, since I make the port's empty during the installation test, I never caught this issue. Looks like the newest CI version has arguments type restriction for this case.

**Comment by abydahana** [comment]
> Additionally, I see that now the default environment is "development"... Where I can set it to Production?

You can change the environment from ./config.php

**Comment by FlavioSuarez** [comment]
OK, got it! Thank you!

**Comment by FlavioSuarez** [comment]
Now, after I created a new menu on the front-end, I get a new error:

Undefined variable $menus
ROOTPATH\themes\default\header.php at line 22

:-( :-(

I can try on a fresh install, but I think it would be useful to fix these errors...

**Comment by abydahana** [comment]
> Now, after I created a new menu on the front-end, I get a new error:
> 
> Undefined variable $menus ROOTPATH\themes\default\header.php at line 22
> 
> :-( :-(
> 
> I can try on a fresh install, but I think it would be useful to fix these errors...

What do you mean with "created a new menu on the front-end"?

**Comment by FlavioSuarez** [comment]
Hi,

Thank you for your reply.

I installed a fresh new Aksara 5.0.6, just for testing it.
On this 5.0.6 install:
- I created 3 new tables on MySQL
- Created 3 controllers for the 3 new tables
- Set permission to these 3 new controllers to SuperAdmin group
- Went to menu management, created a menu placed on "Header" (not on "Sidebar" placement) and created 1 menu item for each controller
- Found the problem of missing Twig files on the default (front end) theme (issue #464)
- Solved the problem copying all the missing Twig files from the back-end theme
- All working fine

Then 5.1.0 was released. 
Following your instructions on this same issue, I copied the 5.1.0 files over the 5.0.6 files
All working fine with the 3 menu items on the header. It seemed ok, at least...
Created 2 new tables on the database
Created a 4th controller
Set permission to this 4th controller to the SuperAdmin group
Went to menu management, and on the menu placed on "Header" (not on "Sidebar" placement) I created a 4th menu item for the 4th controller and saved.

But then, going to the main page and refreshing it throws this error... 
:-(

**Comment by abydahana** [comment]
Looks like the menus are well added. I will try to copy these serialized menus to make sure it's causing the errors. Otherwise the error came from elsewhere. Will inspect it tomorrow.

**Comment by FlavioSuarez** [comment]
Great!

I went back to menus management, refreshed it (CTRL + F5),  and the page remains working.
But when I go to the home page (front-end) the error happens. Even when I open the home page on a new tab...

**Comment by FlavioSuarez** [comment]
I changed the menu placement to "Sidebar" and this brakes the back-end!
I get this error:

Undefined variable $meta
ROOTPATH\themes\backend\layout.php at line 4

The only way to get back the back-end was changing the placement directly on the database, from "sidebar" to "header"...

**Comment by abydahana** [comment]
> Hi,
> 
> Thank you for your reply.
> 
> I installed a fresh new Aksara 5.0.6, just for testing it. On this 5.0.6 install:
> 
> * I created 3 new tables on MySQL
> * Created 3 controllers for the 3 new tables
> * Set permission to these 3 new controllers to SuperAdmin group
> * Went to menu management, created a menu placed on "Header" (not on "Sidebar" placement) and created 1 menu item for each controller
> * Found the problem of missing Twig files on the default (front end) theme (issue [Missing twig files on the default (front end) theme #464](https://github.com/abydahana/aksara/issues/464))
> * Solved the problem copying all the missing Twig files from the back-end theme
> * All working fine
> 
> Then 5.1.0 was released. Following your instructions on this same issue, I copied the 5.1.0 files over the 5.0.6 files All working fine with the 3 menu items on the header. It seemed ok, at least... Created 2 new tables on the database Created a 4th controller Set permission to this 4th controller to the SuperAdmin group Went to menu management, and on the menu placed on "Header" (not on "Sidebar" placement) I created a 4th menu item for the 4th controller and saved.
> 
> But then, going to the main page and refreshing it throws this error... :-(
> 
> On the menus table, the serialized_data column has this content (formatted): [{ "order": 0, "children": [] }, { "id": "1", "icon": "mdi mdi-view-list", "label": "Products", "slug": "products", "newtab": "0", "order": 1, "children": [] }, { "id": "2", "icon": "mdi mdi-radiobox-blank", "label": "Eventos", "slug": "eventos", "newtab": "0", "order": 2, "children": [] }, { "id": "3", "icon": "mdi mdi-react", "label": "Produtos", "slug": "produtos", "newtab": "0", "order": 3, "children": [] }, { "id": "4", "icon": "mdi mdi-alarm-light", "label": "Agro Solidário", "slug": "agrosolidario", "newtab": "0", "order": 4, "children": [] }]

Fixed #470

**Comment by FlavioSuarez** [comment]
Everything is working again!! Thank you!
